Add change notes, upgrade notes, and release announcement for release 1.10 (#9661)
istio/istio.io: content/en/news/releases/1.10.x/announcing-1.10/upgrade-notes/index.md

---
title: Upgrade Notes

> description: Important changes to consider when upgrading to Istio 1.10.0.

Reviewer: Can you please add a publish date? See https://github.com/istio/tools/pull/1509

weight: 20
---

When you upgrade from Istio 1.9 to Istio 1.10, you need to consider the changes on this page.
These notes detail the changes which purposefully break backwards compatibility with Istio 1.9.
The notes also mention changes which preserve backwards compatibility while introducing new behavior.
Changes are only included if the new behavior would be unexpected to a user of Istio 1.9.

## Inbound Forwarding Configuration

The behavior of inbound forwarding will be modified in the near future. While this change is not enabled
by default in Istio 1.10, it can be enabled today by configuring the `PILOT_ENABLE_INBOUND_PASSTHROUGH=true` environment

> variable in Istiod.

Reviewer 2: I think this is not correct?  The blog said it is enabled: https://preliminary.istio.io/latest/blog/2021/upcoming-networking-changes/#future-behavior

Author: Changed, thanks, think the release note had old info

> The behavior of inbound forwarding will be modified in the near future. While this change is not enabled
> by default in Istio 1.10, it can be enabled today by configuring the `PILOT_ENABLE_INBOUND_PASSTHROUGH=true` environment
> variable in Istiod.

Reviewer 2: ```suggestion
The behavior of inbound forwarding will be modified in the near future. While this change is enabled
by default in Istio 1.10, it can be disabled today by configuring the `PILOT_ENABLE_INBOUND_PASSTHROUGH=false` environment
variable in Istiod.
```

Previously, requests would be forwarded to `localhost`. This leads to two important differences compared to running applications
without Istio:

* Applications that bind to `localhost` will be exposed to external pods.
* Applications that bind to `<POD_IP>` will not be exposed to external pods.

The latter is a common source of friction when adopting Istio, in particular with stateful services where this is common.

The new behavior instead forwards the request as is. This matches the behavior a user would see without Istio installed.
However, as a result, applications that have come to rely on `localhost` being exposed externally by Istio may stop working.

> To help detect these situations, we have added a check to find pods that will be impacted. You can run the istioctl
> experimental precheck command to get a report of any pods binding to lo on a port exposed in a Service. This command is

Reviewer 2: ```suggestion
To help detect these situations, we have added a check to find pods that will be impacted. You can run the `istioctl
experimental precheck` command to get a report of any pods binding to lo on a port exposed in a Service. This command is
```

Error [IST0143] (Pod echo-local-849647c5bd-g9wxf.default) Port 7070 is exposed in a Service but listens on localhost. It will not be exposed to other pods.
Error: Issues found when checking the cluster. Istio may not be safe to install or upgrade.
See https://istio.io/latest/docs/reference/config/analysis for more information about causes and resolutions.
{{< /text >}}

Regardless of Istio version, the behavior can be explicitly controlled by the `Sidecar`.
For example, to configure the 9080 port to explicitly be sent to localhost:

{{< text yaml >}}
apiVersion: networking.istio.io/v1beta1
kind: Sidecar
metadata:
  name: ratings
spec:
  workloadSelector:
    labels:
      app: ratings
  ingress:
  - port:
      number: 9080
      protocol: HTTP
      name: http
    defaultEndpoint: 127.0.0.1:9080
{{< /text >}}

## Sidecar Injector Changes

The logic to determine if a pod requires sidecar injection or not has been updated to make use of
new Kubernetes features. Previously, the webhook was triggered at a coarse grain level, selecting any
pods in a namespace with a matching `istio-injection=enabled` label.

This has two limitations:

> * Opting out individual pods with the `sidecar.istio.io/inject` label would still trigger the webhook,

Reviewer 2: Is this an annotation (instead of label)?

Author: Yep, fixed, thanks

> * Opting out individual pods with the `sidecar.istio.io/inject` label would still trigger the webhook,

Reviewer 2: ```suggestion
* Opting out individual pods with the `sidecar.istio.io/inject` annotation would still trigger the webhook,
```

  only to be filtered out by Istio. This can have the unexpected impact of adding a dependency on Istio
  when one is not expected.

* There is no way to opt-in an individual pod, with `sidecar.istio.io/inject`, without enabling injection
  for the entire namespace.

These limitations have both been resolved. As a result, additional pods may be injected that were not in previous versions,

> if they exist in a namespace without an `istio-injection` label set but have the `sidecar.istio.io/inject` label set on the pod.

Reviewer 2: ```suggestion
if they exist in a namespace without an `istio-injection` label set but have the `sidecar.istio.io/inject` annotation set to `true` on the pod.
```

---

> title: Upgrade Notes

Reviewer 3: ```suggestion
title: Istio 1.10 Upgrade Notes
```

description: Important changes to consider when upgrading to Istio 1.10.0.

> publishdate: 2021-05-18

Reviewer 3: ```suggestion
linktitle: 1.10 Upgrade Notes
publishdate: 2021-05-18
```
